django_mesh/models.py

```diff
@@ -40,25 +49,63 @@ class _Abstract(models.Model):     #microblog compatible.
     title = models.CharField(max_length=140, unique=True)
     text = models.TextField(default='')
     rendered_text = models.TextField(default='', blank=True)
+    CONVERSION = Choices(
+        (0, 'PLAINTEXT', 'Plaintext',),
+        (1, 'MARKDOWN', 'Markdown',),
+        (2, 'TEXTILE', 'Textile'),
+        )
+
+    conversion      = models.IntegerField(max_length=1, default=CONVERSION.PLAINTEXT, choices=CONVERSION)
+
+    def render(self, *args, **kwargs):
+
+        self.rendered_text = markdown.markdown(self.text)
+
+        if self.conversion == self.CONVERSION.PLAINTEXT:
+            self.rendered_text = escape(self.text)
+
+        elif self.conversion == self.CONVERSION.TEXTILE:
+            self.rendered_text = textile.textile(self.text)
+
+
+        soup = BeautifulSoup(self.rendered_text)
+
+        url_list = re.findall(url_getter(self.rendered_text), self.rendered_text) # grabs all urls in plaintext
+        url_list2 = []
 
-    def get_oembed_markup(self, matchobj):
-        gd = matchobj.groupdict('')
-        return '%(spacing)s<a href="%(url)s">%(url)s</a>' % gd
+        for each in url_list:
+            each_soup = BeautifulSoup(each) # change into BS object
+            for each_url in each_soup: # changes into BSnavigatable string
+                each_url = each_url.wrap(each_soup.new_tag('a', href=each_url)) # anchored url; have to figure out how to put this into soup
+                url_list2.append(each_url)
 
-    def render(self):
-        #TODO: strip out dangerous HTML attributes, only allow basic formatting tags
-        self.rendered_text = oembed_regex.sub(self.get_oembed_markup, self.text)
+        text = self.rendered_text
+        for i in range(0, len(url_list)-1):
+            text = text.replace(url_list[i], str(url_list2[i]))
```

Comment:
This will replace all instances of a URL. So if a URL appears multiple times, such as once in a text node and once as an href attribute, the href attribute will get replaced, resulting in invalid markup.